wagn/lib/kaminari/helpers/tags.rb

```diff
@@ -0,0 +1,130 @@
+module Kaminari
+  module Helpers
+    PARAM_KEY_BLACKLIST = :authenticity_token, :commit, :utf8, :_method
+
+    # A tag stands for an HTML tag inside the paginator.
+    # Basically, a tag has its own partial template file, so every tag can be
+    # rendered into String using its partial template.
+    #
+    # The template file should be placed in your app/views/kaminari/ directory
+    # with underscored class name (besides the "Tag" class. Tag is an abstract
+    # class, so _tag parital is not needed).
+    #   e.g.)  PrevLink  ->  app/views/kaminari/_prev_link.html.erb
+    #
+    # When no matching template were found in your app, the engine's pre
+    # installed template will be used.
+    #   e.g.)  Paginator  ->  $GEM_HOME/kaminari-x.x.x/app/views/kaminari/_paginator.html.erb
+    class Tag
+      def initialize(template, options = {}) #:nodoc:
+        @template, @options = template, options.dup
+        @param_name = @options.delete(:param_name) || Kaminari.config.param_name
+        @theme = @options.delete(:theme)
+        @views_prefix = @options.delete(:views_prefix)
+        @params = template.params.except(*PARAM_KEY_BLACKLIST).merge(@options.delete(:params) || {})
+      end
+
+      def to_s(locals = {}) #:nodoc:
+        @template.render :partial => partial_path, :locals => @options.merge(locals), :formats => [:html]
+      end
+
+      def page_url_for(page)
+        p = params_for(page).merge(:only_path => true)
+        #view"=>"history", "controller"=>"card", "action"=>"read", "format"=>"html", "id"=>"Home", "page"=>"1",
+        "#{p['id']}?view=#{p['view']}&page=#{p['page']}"
+      end
```

Comment:
This file is a copy from the kaminari gem. The url_for method doesn't work in the original code. Maybe it has something to do with our new engine routes and we can get rid of this when the PR https://github.com/amatsuda/kaminari/pull/636 gets merged.